packages/elevenlabs/src/elevenlabs-transcription-model.ts

```diff
@@ -0,0 +1,318 @@
+
+    // Parse provider options
+    const elevenlabsOptions = parseProviderOptions({
+      provider: 'elevenlabs',
+      providerOptions,
+      schema: ElevenLabsProviderOptionsSchema,
+    });
+
+    // Create form data with base fields
+    const formData = new FormData();
+    const blob =
+      audio instanceof Uint8Array
+        ? new Blob([audio])
+        : new Blob([convertBase64ToUint8Array(audio)]);
+
+    formData.append('model_id', this.modelId);
+    formData.append('file', new File([blob], 'audio', { type: mediaType }));
+
+    // Add provider-specific options
+    if (elevenlabsOptions) {
+      const transcriptionModelOptions: ElevenLabsTranscriptionModelOptions = {
+        language_code: elevenlabsOptions.languageCode,
+        tag_audio_events: elevenlabsOptions.tagAudioEvents,
+        num_speakers: elevenlabsOptions.numSpeakers,
+        timestamps_granularity: elevenlabsOptions.timestampsGranularity,
+        diarize: elevenlabsOptions.diarize,
+        additional_formats: elevenlabsOptions.additionalFormats,
+        file_format: elevenlabsOptions.file_format,
+      };
+
+      for (const key in transcriptionModelOptions) {
+        const value =
+          transcriptionModelOptions[
+            key as keyof ElevenLabsTranscriptionModelOptions
+          ];
+        if (value !== undefined) {
+          formData.append(key, String(value));
+        }
+      }
+    }
+
+    return {
+      formData,
+      warnings,
+    };
+  }
+
+  async doGenerate(
+    options: ElevenLabsTranscriptionCallOptions,
+  ): Promise<Awaited<ReturnType<TranscriptionModelV1['doGenerate']>>> {
+    const currentDate = this.config._internal?.currentDate?.() ?? new Date();
+    const { formData, warnings } = this.getArgs(options);
+
+    const {
+      value: response,
+      responseHeaders,
+      rawValue: rawResponse,
+    } = await postFormDataToApi({
+      url: this.config.url({
+        path: '/v1/speech-to-text',
+        modelId: this.modelId,
+      }),
+      headers: combineHeaders(this.config.headers(), options.headers),
+      formData,
+      failedResponseHandler: elevenlabsFailedResponseHandler,
+      successfulResponseHandler: createJsonResponseHandler(
+        elevenlabsTranscriptionResponseSchema,
+      ),
+      abortSignal: options.abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    return {
+      text: response.text,
+      segments:
+        response.words?.map(word => ({
+          text: word.text,
+          startSecond: word.start ?? 0,
+          endSecond: word.end ?? 0,
+        })) ?? [],
+      language: response.language_code,
+      durationInSeconds: response.words?.at(-1)?.end ?? undefined,
+      warnings,
+      response: {
+        timestamp: currentDate,
+        modelId: this.modelId,
+        headers: responseHeaders,
+        body: rawResponse,
+      },
+    };
+  }
+}
+
+const elevenlabsTranscriptionResponseSchema = z.object({
```

Comment:
Response schemas are only for our validation and should be minimal to reduce package size (no `describe`, no unused properties, also prefer `nullish` over `optional`)